### PR TITLE
Clarify MinAvailable approval semantics.

### PR DESCRIPTION
**Pull request by mml**
Fixes doc comments to be in line with #33143

---

This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/33148)

### CODE REVIEW

File: pkg/apis/policy/types.go [hunk 1]
```diff
@@ -26,6 +26,10 @@ import (
 type PodDisruptionBudgetSpec struct {
 	// The minimum number of pods that must be available simultaneously.  This
 	// can be either an integer or a string specifying a percentage, e.g. "28%".
+	// This minimum is applicable at all times.  Specifically, when the
```
Comment (davidopp) [inline pkg/apis/policy/types.go]: This is getting kinda verbose. How about replacing the whole comment with

```
An eviction is allowed if at least "minAvailable" pods selected by "selector" will
still be available after the eviction, i.e. even in the absence of the evicted pod.
So for example you can prevent all voluntary evictions by specifying "100%".
```

**Review by davidopp — CHANGES_REQUESTED** [review]

**Comment by mml** [comment]
@davidopp PTAL

**Comment by davidopp** [comment]
LGTM

**Comment by mml** [comment]
@k8s-bot verify test this #IGNORE

**Comment by mml** [comment]
@k8s-bot gke e2e test this #34483

**Pull request merged**